### ISSUE TITLE
[Game Crash] "Access Violation"

### ISSUE THREAD

**Issue by ScottBrenner** [TeamRizu/OutFox#72]
1. What you were doing before it crashed and what action made the game to crash.
Mapping inputs inputs for my [L-TEK pad](https://www.maty-taneczne.pl/shop/dance-mat-ddr-usb-ex-pro/), specifically the [START / SELECT Buttons](https://www.maty-taneczne.pl/shop/start-select-buttons/). Pressed Escape on my keyboard to save changes and the game crashed.
2. Upload the crashinfo.txt (GameFolder/crashinfo.txt)
<details>
<summary>crashinfo.txt</summary>
<br>

```
StepMania5.3-git-24199dfb28 crash report (build 24199dfb28, 20200814 @ 17:32:29)
--------------------------------------

Crash reason: Access Violation

Thread: main
        GameSoundManager.cpp:584 0.947574, delta 0.017000
        GameSoundManager.cpp:584 0.947574, delta 0.016000
        GameSoundManager.cpp:584 0.947574, delta 0.017000
        FontManager.cpp:63 FontManager::UnloadFont(/Appearance/Themes/default/Fonts/_ibm plex sans semibold 36px.ini).
        FontManager.cpp:63 FontManager::UnloadFont(/Appearance/Themes/default/Fonts/_plex sans condensed medium 24px.ini).
Thread: Decode thread
        RageSoundDriver_Generic_Software.cpp:204 Processing the sound while buffers are available.
        RageSoundDriver_Generic_Software.cpp:204 Processing the sound while buffers are available.
        RageSoundDriver_Generic_Software.cpp:204 Processing the sound while buffers are available.
        RageSoundDriver_Generic_Software.cpp:204 Processing the sound while buffers are available.
        RageSoundDriver_Generic_Software.cpp:204 Processing the sound while buffers are available.
Thread: Mixer thread
Thread: Music thread
        RageFileDriverDirectHelpers.cpp:229 C:/Games/StepMania 5.3 Outfox/Appearance/Themes/default/*
        RageFileDriverDirectHelpers.cpp:229 C:/Games/StepMania 5.3 Outfox/Appearance/Themes/default/Sounds/*
        RageFileDriverDirectHelpers.cpp:229 C:/Games/StepMania 5.3 Outfox/Appearance/Themes/default/*
        RageFileDriverDirectHelpers.cpp:229 C:/Games/StepMania 5.3 Outfox/Appearance/Themes/_fallback/Sounds/*
Thread: Worker thread (MemoryCardWorker)
Thread: Worker thread (/@mc1int/)
Thread: Worker thread (/@mc2int/)
0070ddfa: StepMania!00660000
0070de0f: StepMania!00660000
0070de0f: StepMania!00660000
0070de0f: StepMania!00660000
0070de0f: StepMania!00660000
0070de0f: StepMania!00660000
0070de0f: StepMania!00660000
0070de0f: StepMania!00660000
0070de0f: StepMania!00660000
0070de0f: StepMania!00660000
008b9ccb: StepMania!00660000
008b4c3a: StepMania!00660000
007034f9: StepMania!00660000
76fadcb6: ntdll!RtlFreeHeap [76f70000+3dc70+46]
00cfb6b1: StepMania!00660000
007c6d31: StepMania!00660000
00842873: StepMania!00660000
00b2449c: StepMania!00660000
00aa0f1b: StepMania!00660000
0076f1e5: StepMania!00660000
0076f726: StepMania!00660000
0076e65d: StepMania!00660000
00a46818: StepMania!00660000
00a2725f: StepMania!00660000
0077140e: StepMania!00660000
0089add1: StepMania!00660000
0089adea: StepMania!00660000
006efd00: StepMania!00660000
006f003c: StepMania!00660000
006f8e72: StepMania!00660000
00d0a845: StepMania!00660000
00d0a845: StepMania!00660000
00d0a795: StepMania!00660000
00cd7989: StepMania!00660000
76fd7bf4: ntdll!RtlGetAppContainerNamedObjectPath [76f70000+67b40+b4]

Static log:
StepMania5.3-git-24199dfb28
Compiled 20200814 @ 17:32:29 (build 24199dfb28)
Log starting 2020-08-15 17:14:52
Loading window: win32
Windows 10
Memory: 2047mb total, 4095mb swap (4095mb swap avail)
Video driver: NVIDIA Quadro 1000M [NVIDIA]
              21.21.13.7755, 7-13-2017 [pci\ven_10de&dev_0dfa&subsys_21d117aa]
Drive: "        OCZ-VERTEX4" Driver: storahci DMA: N/A
Sound device 0: VW32L HDTV40A-1 (NVIDIA High De, 10.0, MID 1, PID 100 
Sound device 1: Speakers (High Definition Audio, 0.0, MID 1, PID 65535 
WaveOut software mixing at 44100 hz
Sound driver: WaveOut
Lights driver: SystemMessage
Lights driver: Export
WARNING: Unmatched 3 in "0003"
WARNING: Unmatched 3 in "0300"
WARNING: Unmatched 3 in "0300"
WARNING: Unmatched 3 in "3000"
WARNING: Unmatched 3 in "3000"
WARNING: Unmatched 3 in "0300"
WARNING: Unmatched 3 in "3000"
WARNING: Unmatched 3 in "0300"
WARNING: Unmatched 3 in "0003"
WARNING: Unmatched 3 in "0030"
WARNING: Unmatched 3 in "0003"
WARNING: Unmatched 3 in "0030"
WARNING: Unmatched 3 in "1003"
WARNING: Unmatched 3 in "0031"
WARNING: Unmatched 3 in "0310"
WARNING: Unmatched 3 in "0003"
WARNING: Unmatched 3 in "0030"
WARNING: Unmatched 3 in "0003"
WARNING: Unmatched 3 in "0030"
WARNING: Unmatched 3 in "0300"
WARNING: Unmatched 3 in "0003"
Video renderers: 'glad,opengl,gles2'
Graphics Manager:  OpenGL Vendor: NVIDIA Corporation
Graphics Manager:  OpenGL Renderer: Quadro 1000M/PCIe/SSE2
Graphics Manager:  OpenGL Version: 4.5.0 NVIDIA 377.55
Graphics Manager:  OpenGL Max texture size: 16384
Graphics Manager:  OpenGL Texture units: 4
Graphics Manager:  OpenGL Extensions:
  GL_AMD_multi_draw_indirect
  GL_ARB: ES2_compatibility, ES3_1_compatibility, ES3_2_compatibility, ES3_compatibility, arrays_of_arrays, 
    base_instance, blend_func_extended, buffer_storage, cl_event, clear_buffer_object, clear_texture, 
    clip_control, color_buffer_float, compatibility, compressed_texture_pixel_storage, compute_shader, 
    compute_variable_group_size, conditional_render_inverted, conservative_depth, copy_buffer, copy_image, 
    cull_distance, debug_output, depth_buffer_float, depth_clamp, depth_texture, derivative_control, 
    direct_state_access, draw_buffers, draw_buffers_blend, draw_elements_base_vertex, draw_indirect, 
    draw_instanced, enhanced_layouts, explicit_attrib_location, explicit_uniform_location, 
    fragment_coord_conventions, fragment_layer_viewport, fragment_program, fragment_program_shadow, 
    fragment_shader, framebuffer_no_attachments, framebuffer_object, framebuffer_sRGB, geometry_shader4, 
    get_program_binary, get_texture_sub_image, gl_spirv, gpu_shader5, gpu_shader_fp64, gpu_shader_int64, 
    half_float_pixel, half_float_vertex, imaging, indirect_parameters, instanced_arrays, internalformat_query, 
    internalformat_query2, invalidate_subdata, map_buffer_alignment, map_buffer_range, multi_bind, 
    multi_draw_indirect, multisample, multitexture, occlusion_query, occlusion_query2, parallel_shader_compile, 
    pipeline_statistics_query, pixel_buffer_object, point_parameters, point_sprite, program_interface_query, 
    provoking_vertex, query_buffer_object, robust_buffer_access_behavior, robustness, sample_shading, 
    sampler_objects, seamless_cube_map, separate_shader_objects, shader_atomic_counter_ops, 
    shader_atomic_counters, shader_bit_encoding, shader_draw_parameters, shader_group_vote, 
    shader_image_load_store, shader_image_size, shader_objects, shader_precision, shader_storage_buffer_object, 
    shader_subroutine, shader_texture_image_samples, shader_texture_lod, shading_language_100, 
    shading_language_420pack, shading_language_include, shading_language_packing, shadow, sparse_buffer, 
    sparse_texture, stencil_texturing, sync, tessellation_shader, texture_barrier, texture_border_clamp, 
    texture_buffer_object, texture_buffer_object_rgb32, texture_buffer_range, texture_compression, 
    texture_compression_bptc, texture_compression_rgtc, texture_cube_map, texture_cube_map_array, texture_env_add, 
    texture_env_combine, texture_env_crossbar, texture_env_dot3, texture_float, texture_gather, 
    texture_mirror_clamp_to_edge, texture_mirrored_repeat, texture_multisample, texture_non_power_of_two, 
    texture_query_levels, texture_query_lod, texture_rectangle, texture_rg, texture_rgb10_a2ui, texture_stencil8, 
    texture_storage, texture_storage_multisample, texture_swizzle, texture_view, timer_query, transform_feedback2, 
    transform_feedback3, transform_feedback_instanced, transform_feedback_overflow_query, transpose_matrix, 
    uniform_buffer_object, vertex_array_bgra, vertex_array_object, vertex_attrib_64bit, vertex_attrib_binding, 
    vertex_buffer_object, vertex_program, vertex_shader, vertex_type_10f_11f_11f_rev, vertex_type_2_10_10_10_rev, 
    viewport_array, window_pos
  GL_ATI: draw_buffers, texture_float, texture_mirror_once
  GL_EXTX_framebuffer_mixed_formats
  GL_EXT: Cg_shader, abgr, bgra, bindable_uniform, blend_color, blend_equation_separate, blend_func_separate, 
    blend_minmax, blend_subtract, compiled_vertex_array, depth_bounds_test, direct_state_access, draw_buffers2, 
    draw_instanced, draw_range_elements, fog_coord, framebuffer_blit, framebuffer_multisample, 
    framebuffer_multisample_blit_scaled, framebuffer_object, framebuffer_sRGB, geometry_shader4, 
    gpu_program_parameters, gpu_shader4, import_sync_object, multi_draw_arrays, packed_depth_stencil, 
    packed_float, packed_pixels, pixel_buffer_object, point_parameters, polygon_offset_clamp, provoking_vertex, 
    rescale_normal, secondary_color, separate_shader_objects, separate_specular_color, shader_image_load_store, 
    shader_integer_mix, shadow_funcs, stencil_two_side, stencil_wrap, texture3D, texture_array, 
    texture_buffer_object, texture_compression_dxt1, texture_compression_latc, texture_compression_rgtc, 
    texture_compression_s3tc, texture_cube_map, texture_edge_clamp, texture_env_add, texture_env_combine, 
    texture_env_dot3, texture_filter_anisotropic, texture_integer, texture_lod, texture_lod_bias, 
    texture_mirror_clamp, texture_object, texture_sRGB, texture_sRGB_decode, texture_shared_exponent, 
    texture_storage, texture_swizzle, timer_query, transform_feedback2, vertex_array, vertex_array_bgra, 
    vertex_attrib_64bit, window_rectangles
  GL_IBM: rasterpos_clip, texture_mirrored_repeat
  GL_KHR: blend_equation_advanced, context_flush_control, debug, no_error, robust_buffer_access_behavior, 
    robustness
  GL_KTX_buffer_region
  GL_NVX: conditional_render, gpu_memory_info, multigpu_info
  GL_NV: ES1_1_compatibility, ES3_1_compatibility, alpha_to_coverage_dither_control, bindless_multi_draw_indirect, 
    bindless_multi_draw_indirect_count, blend_equation_advanced, blend_square, command_list, compute_program5, 
    conditional_render, copy_depth_to_color, copy_image, depth_buffer_float, depth_clamp, draw_texture, 
    explicit_multisample, fence, float_buffer, fog_distance, fragment_program, fragment_program2, 
    fragment_program_option, framebuffer_multisample_coverage, geometry_shader4, gpu_program4, gpu_program4_1, 
    gpu_program5, gpu_program5_mem_extended, gpu_program_fp64, gpu_shader5, half_float, 
    internalformat_sample_query, light_max_exponent, multisample_coverage, multisample_filter_hint, 
    occlusion_query, packed_depth_stencil, parameter_buffer_object, parameter_buffer_object2, path_rendering, 
    pixel_data_range, point_sprite, primitive_restart, register_combiners, register_combiners2, 
    shader_atomic_counters, shader_atomic_float, shader_buffer_load, shader_storage_buffer_object, 
    shader_thread_group, texgen_reflection, texture_barrier, texture_compression_vtc, texture_env_combine4, 
    texture_multisample, texture_rectangle, texture_shader, texture_shader2, texture_shader3, transform_feedback, 
    transform_feedback2, uniform_buffer_unified_memory, vertex_array_range, vertex_array_range2, 
    vertex_attrib_integer_64bit, vertex_buffer_unified_memory, vertex_program, vertex_program1_1, vertex_program2, 
    vertex_program2_option, vertex_program3
  GL_S3_s3tc
  GL_SGIS: generate_mipmap, texture_lod
  GL_SGIX: depth_texture, shadow
  GL_SUN_slice_accum
  GL_WIN_swap_hint
  WGL_EXT_swap_control
GLAD Fullscreen 1190x670 32 color 32 texture 60Hz Vsync SmoothLines
Disconnected: Mata taneczna L-TEK
Auto-mapping all joysticks.
Players joined: P1, P2
Top Screen: ScreenOptionsMapping
Language: en
Current renderer: OpenGL
Theme: default

Partial log:
02:24.811: OutFox Engine: Opening Controller with GamePad ID 0
02:24.811: OutFox Engine: Opening Joystick Subsystem with device id 1, instance ID -1
02:24.811: Outfox Engine: Joystick/Controller qualified name ((null))
02:24.811: Outfox Engine: Joystick/Controller device id 1, instance ID -1, Device name ((null))
02:24.811: Outfox Engine: Axes -1, Buttons -1, Hats -1, Balls -1
02:24.811: Outfox Engine: Instance id -1
02:24.811: Outfox Engine: Guid 00000000000000000000000000000000
02:24.811: OutFox Engine: Haptic Effects: -1
02:24.811: OutFox Engine: Haptic Query: 0
02:24.811: OutFox Engine: Haptic Rumble Init: Haptic: Invalid haptic device identifier

-- End of report
```
</details>

**Additional information**

Had just updated from StepMania 5.3 Alpha 4.5.8 and seemed to have lost the START / SELECT Button mappings, went into the settings to re-map and the game crashed. Have not since been able to map these buttons (not saving?). Additionally, each time I start the game now I have to unplug/replug my L-TEK pad for any inputs to register..

> StepMania5.3-git-24199dfb28
> Crash reason: Access Violation
> Windows 10
> Memory: 2047mb total, 4095mb swap (4095mb swap avail)
> Video driver: NVIDIA Quadro 1000M [NVIDIA]

**Comment by KaeMame** [comment]
While testing something else, I just reproduced the issue on Linux (it didn't happen before, I guess it's kind of random). I see no crashinfo.txt in my Logs folder, so here's log.txt
[log.txt](https://github.com/TeamRizu/OutFox/files/5080661/log.txt)

**Comment by Scraticus** [comment]
This issue is fixed in the next update, we are just running through some testing.

**Comment by Scraticus** [comment]
Fixed in 4.8.0, Thanks for reporting!

**Comment by ScottBrenner** [comment]
Confirmed!
